db/log_writer.cc

```diff
@@ -89,8 +93,15 @@ Status Writer::EmitPhysicalRecord(RecordType t, const char* ptr, size_t n) {
 
   // Compute the crc of the record type and the payload.
   uint32_t crc = crc32c::Extend(type_crc_[t], ptr, n);
-  crc = crc32c::Mask(crc);                 // Adjust for storage
-  EncodeFixed32(buf, crc);
+  if (db_options_->recycle_log_files) {
```

Comment:
If it is the only place db_options_ is used, I would suggest we directly pass recycle_log_files as a parameter. It will make write unit tests in single modules easier.